Wikipedia/assets/codemirror/codemirror-index.html

```diff
@@ -665,35 +665,39 @@
         console.log('decreaseIndentDepth')
       },
       textSize: (newSize) => {
-        if (newSize === 'small') {
-          if (selectionPayloadForButton('bigTextSize')) {
-            expandTagSelectionAndToggleMarkup('bigTextSize', '<big>', '</big>')  
-          }
-          if (selectionPayloadForButton('smallTextSize')) {
-            // If after toggle above the selection is still inside a 'small' (ie 'big' was nested in small before it was toggled off) expand the selection to encompass the small
-            expandTagSelection('smallTextSize')
-          }
-          expandTagSelectionAndToggleMarkup('smallTextSize', '<small>', '</small>')
-          return
-        }
-        if (newSize === 'big') {
-          if (selectionPayloadForButton('smallTextSize')) {
-            expandTagSelectionAndToggleMarkup('smallTextSize', '<small>', '</small>')  
-          }
-          if (selectionPayloadForButton('bigTextSize')) {
-            // If after toggle above the selection is still inside a 'big' (ie 'small' was nested in big before it was toggled off) expand the selection to encompass the big
-            expandTagSelection('bigTextSize')
-          }
-          expandTagSelectionAndToggleMarkup('bigTextSize', '<big>', '</big>')  
-          return
-        }
-        if (newSize === 'normal') {
-          if (selectionPayloadForButton('smallTextSize')) {
-            expandTagSelectionAndToggleMarkup('smallTextSize', '<small>', '</small>')  
-          }
-          if (selectionPayloadForButton('bigTextSize')) {
-            expandTagSelectionAndToggleMarkup('bigTextSize', '<big>', '</big>')  
-          }
+        const selectedButton = selectionPayloadForButton('textSize')
```

Comment:
It looks like the native side still needs to be updated to handle the size messages from JS land:
![aaaaa mov](https://user-images.githubusercontent.com/3143487/50936645-5e4d9900-1425-11e9-9f14-99c97ee15ed8.gif)

I double-checked and it looked the size message is making to native land correctly.

Checking out a pre-native-refactor commit (1da401c0b) the JS looks to have been behaving itself, so I'm curious if fixing the native issue above suffices. The reason I wonder this is these new JS changes broke one key thing the old JS structure was trying to do: I made an assumption a user would be more likely to intend the behavior in the first animation below than the second one. I'm curious if you think the same. (side note: the repeated calls to `selectionPayloadForButton` were intentional part of the old approach - though i'm sure they looked sloppy 🤪)

Before this PR (JS seems ok though we still had native interface switching back to `Text formatting` issue):
![bbbbbbb mov](https://user-images.githubusercontent.com/3143487/50937715-bdada800-1429-11e9-8b9e-05e0eace0cee.gif)

With this PR's changes:
![ccccccccccc mov](https://user-images.githubusercontent.com/3143487/50937710-ba1a2100-1429-11e9-8ca9-0581c8bf9654.gif)

Reply:
Thanks for testing and catching this!

When I was refactoring, I changed the `textSize` message coming from JS - instead of receiving `smallTextSize` and `bigTextSize` as separate messages, I wanted to receive a `textSize` message with information about the size inside `info` (just like what `heading` does with `depth`). However, I didn't update it here so whenever I was telling the text size to change, it was checking whether `smallTextSize` or `bigTextSize` were among the selected buttons, which was never the case anymore. So this is what this change should be fixing - checking whether `textSize` is selected and figuring out the size via `info.size`. I updated it so that each case checks for the `selectedButton`, the way you had it before. If you think there's a better way to do this, let me know or feel free to push an update

The native side changed to fix the pop visible in 1da401c